master/buildbot/statistics/capture.py

```diff
@@ -80,25 +80,74 @@ def consume(self, routingKey, msg):
         send them to the storage backends.
         """
         builder_info = yield self.master.data.get(("builders", msg['builderid']))
-        if self._builder_name == builder_info['name']:
+
+        if self._builder_name_matches(builder_info):
             properties = yield self.master.data.get(("builds", msg['buildid'], "properties"))
-            try:
-                ret_val = self._callback(properties, self._property_name)
-            except KeyError:
-                config.error("CaptureProperty failed."
-                             " The property %s not found for build number %s on builder %s."
-                             % (self._property_name, msg['number'], self._builder_name))
-            context = self._defaultContext(msg)
-            series_name = '%s-%s' % (self._builder_name, self._property_name)
-            post_data = {
-                "name": self._property_name,
-                "value": ret_val
-            }
-            yield self._store(post_data, series_name, context)
+            filtered_prop_names = []
+            if self._regex:
+                prop_names = properties.keys()
+                for pn in prop_names:
+                    if re.match(self._property_name, pn):
+                        filtered_prop_names.append(pn)
```

Comment:
you can rewrite it:

    filtered_prop_names = [pn for pn in properties if re.match(self._property_name, pn]